### DISCUSSION TITLE
Getting current user from .Net in Uno/UWP project

### DISCUSSION THREAD

**Discussion by ghost** [MarimerLLC/csla#2745] [Questions] [answered]
In my UWP project under the Uno platform, I am trying to get current user data from .Net.

In App.xaml.cs, OnLaunched, I execute:
    Csla.ApplicationContext.AuthenticationType = "Windows";
    var principal = Csla.ApplicationContext.User;

When I examine principal.Identity, Name and AuthenticationType are both "".  What am I overlooking?  On the premise that it is something very fundamental, I'm embarrassed to have to ask.

**Comment by rockfordlhotka** [comment, marked as answer]
It is up to the host (WinForms, WPF, Uno, etc.) to set the current user. In WinForms/WPF CSLA defaults to getting the user from `System.Threading.Thread.CurrentThread.CurrentPrincipal` (or something along that line).

Or at least it _used to do this_. A few years ago, Microsoft changed how things work, so the current thread is no longer a reliable source for the user identity. Grumble.

So, assuming CSLA 5 here, the _default_ behavior for CSLA is still to try to get the principal from the thread.

https://github.com/MarimerLLC/csla/blob/dfaf1520e0e5acc05ee910b4f88703aafa137822/Source/Csla.Shared/Core/ApplicationContextManager.cs#L45

However, if your app references the `Csla.Xaml` package, the code _attempts_ to keep the thread set correctly, but ultimately _relies_ on a value maintained in a static field.

https://github.com/MarimerLLC/csla/blob/dfaf1520e0e5acc05ee910b4f88703aafa137822/Source/Csla.Xaml.Shared/ApplicationContextManager.cs#L24

You should see what context manager is being used in your app. You can do this by examining the ContextManager property of the ApplicationContext. If it is one of the types from `Csla.Core` then you need to set the property (once, on app startup) to the context manager from `Csla.Xaml`.

Or, if you aren't using the `Csla.Xaml` package, then copy that class from `Csla.Xaml.Shared` into your project and have ApplicationContext use it.

**Reply by ghost** [reply]
Thank you for the prompt reply.  You are incredible with your help!
